components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -468,6 +469,12 @@ def validate_target_segment_size(cls, field):
             raise ValueError("target_segment_size must be greater than 0")
         return field
 
+    @validator("compression_level")
+    def validate_compression_level(cls, field):
+        if field < 1 or field > 9:
```

Comment:
This is just personal preference, so you can disagree: You could write it as

```suggestion
        if field < 1 or 9 < field:
```

which feels more indicative of a test for `field` being outside the range.